.github/workflows/therock-repo-diff.yml

```diff
@@ -0,0 +1,67 @@
+name: Commit Diff Report
+
+on:
+  workflow_dispatch:
+    inputs:
+      start_commit:
+        description: 'Start Commit SHA or Workflow ID'
+        required: true
+        type: string
+      end_commit:
+        description: 'End Commit SHA or Workflow ID'
+        required: true
+        type: string
+      workflow_mode:
+        description: 'Use Workflow Mode (extract commits from workflow IDs instead of direct commit comparison)'
+        required: false
+        type: boolean
+        default: false
+      find_last_successful:
+        description: 'Workflow name to find last successful run (e.g., "ci_nightly.yml"). If provided, start_commit will be ignored.'
+        required: false
+        type: string
+  workflow_call:
+    inputs:
+      start_commit:
+        description: 'Start Commit SHA or Workflow ID'
+        required: false
+        type: string
+      end_commit:
+        description: 'End Commit SHA or Workflow ID'
+        required: true
+        type: string
+      workflow_mode:
+        description: 'Use Workflow Mode (extract commits from workflow IDs instead of direct commit comparison)'
+        required: false
+        type: boolean
+        default: false
+      find_last_successful:
+        description: 'Workflow name to find last successful run (e.g., "ci_nightly.yml"). If provided, start_commit will be ignored.'
+        required: false
+        type: string
+
+jobs:
+  therock-diff-report:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+
+      - name: Run TheRock Commit Diff Analysis
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
```

Comment:
As stated in an earlier review, let's drop "therock" from these file names and workflow steps.